lncfg/remotesigner.go

```diff
@@ -11,34 +11,91 @@ const (
+	// flag.
+	MigrateWatchOnly bool `long:"migrate-wallet-to-watch-only" description:"If a wallet with private key material already exists, migrate it into a watch-only wallet on first startup. WARNING: This cannot be undone! Make sure you have backed up your seed before you use this flag! All private keys will be purged from the wallet after first unlock with this flag!"`
+
+	// ConnectionCfg holds the connection configuration options that the
+	// watch-only node will use when setting up the connection to the remote
+	// signer.
+	ConnectionCfg
+}
+
+// DefaultRemoteSignerCfg returns the default RemoteSigner config.
+func DefaultRemoteSignerCfg() *RemoteSigner {
+	return &RemoteSigner{
+		Enable:        false,
+		ConnectionCfg: defaultConnectionCfg(),
+	}
 }
 
 // Validate checks the values configured for our remote RPC signer.
 func (r *RemoteSigner) Validate() error {
+	if r.MigrateWatchOnly && !r.Enable {
+		return fmt.Errorf("remote signer: cannot turn on wallet " +
+			"migration to watch-only if remote signing is not " +
+			"enabled")
+	}
+
 	if !r.Enable {
 		return nil
 	}
 
-	if r.Timeout < time.Millisecond {
-		return fmt.Errorf("remote signer: timeout of %v is invalid, "+
-			"cannot be smaller than %v", r.Timeout,
-			time.Millisecond)
+	// Else, we are in outbound mode, so we verify the connection config.
+	err := r.ConnectionCfg.Validate()
+	if err != nil {
+		return fmt.Errorf("remotesigner.%w", err)
 	}
 
-	if r.MigrateWatchOnly && !r.Enable {
-		return fmt.Errorf("remote signer: cannot turn on wallet " +
-			"migration to watch-only if remote signing is not " +
-			"enabled")
+	return nil
+}
+
+// ConnectionCfg holds the configuration options required when setting up a
+// connection to either a remote signer or watch-only node, depending on which
+// side makes the outbound connection.
+//
+//nolint:ll
+type ConnectionCfg struct {
+	RPCHost      string        `long:"rpchost" description:"The RPC host:port of the remote signer. For watch-only nodes, this should be set to the remote signer's RPC host:port."`
+	MacaroonPath string        `long:"macaroonpath" description:"The macaroon to use for authenticating with the remote signer. For watch-only nodes, this should be set to the remote signer's macaroon."`
+	TLSCertPath  string        `long:"tlscertpath" description:"The TLS certificate to use for establishing the remote signer's identity. For watch-only nodes, this should be set to the remote signer's TLS certificate."`
+	Timeout      time.Duration `long:"timeout" description:"The timeout for making the connection to the remote signer. Valid time units are {s, m, h}."`
+}
+
+// defaultConnectionCfg returns the default ConnectionCfg config.
+func defaultConnectionCfg() ConnectionCfg {
+	return ConnectionCfg{
+		Timeout: DefaultRemoteSignerRPCTimeout,
+	}
+}
+
+// Validate checks the values set in the ConnectionCfg config are valid.
+func (c *ConnectionCfg) Validate() error {
+	if c.Timeout < time.Millisecond {
```

Comment:
Q: should we allow setting no timeout? IIRC, when setting this value to zero, the underlying lib will assume no timeout.